integrations/oci/oci-objectstorage/src/main/java/io/helidon/integrations/oci/objectstorage/OciObjectStorageProducer.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.integrations.oci.objectstorage;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+import com.oracle.bmc.auth.AuthenticationDetailsProvider;
+import com.oracle.bmc.objectstorage.ObjectStorage;
+
+
+/**
+ * Class OciObjectStorageProduced.
+ */
+@ApplicationScoped
+public class OciObjectStorageProducer {
+
+    private ObjectStorage objectStorage;
+
+    /**
+     * Constructor for Object Storage.
+     *
+     * @param authenticationDetailsProvider from CDI environment
+     */
+    @Inject
+    public OciObjectStorageProducer(AuthenticationDetailsProvider authenticationDetailsProvider) {
+        this.objectStorage = OciObjectStorage.builder()
+                .authenticationDetailsProvider(authenticationDetailsProvider)
+                .build()
+                .getObjectStorage();
+    }
+
+    /**
+     * Produces Object storage in CDI environment.
+     *
+     * @return the ObjectStorage.
+     */
+    @Produces
```

Comment:
If there is a possibility to use more than one objectstorage from a microservice, then we need to have support for named instances and named configuration to support it.
If we can only have one, then this is sufficient - please validate with OCI docs